[BUG] Non-empty line after one-liner type with requires clause fails assertion

<a id="title"/>

**[T](#title)itle**: Non-empty line after one-liner type with requires clause fails assertion.

<a id="rep"/>
<a id="cpp2"/>

**[M](#rep)inimal reproducer** (<https://cpp2.godbolt.org/z/bfh5bfTj4>):

```Cpp2
t: type requires true = { }
main: () = { }
```

<a id="cmd"/>
<details><summary>
<b><a href="#cmd">C</a>ommands</b>:
</summary>

```bash
cppfront -clean-cpp1 main.cpp2
```
</details>

<a id="actual"/>

**[A](#actual)ctual result and error**:

<a href="#out">O</a>utput.

```output
cppfront: source/cppfront.cpp:523: void cpp2::positional_printer::align_to(cpp2::source_position): Assertion `psource && 0 <= curr_pos.lineno && curr_pos.lineno < std::ssize(psource->get_lines())+1' failed.
```

I have an easy fix for the assertion:
```diff
diff --git a/source/cppfront.cpp b/source/cppfront.cpp
index 5b89964..44cc52e 100644
--- a/source/cppfront.cpp
+++ b/source/cppfront.cpp
@@ -5094,7 +5094,7 @@ public:
                 if (n.requires_clause_expression) {
                     printer.print_cpp2("requires( ", n.requires_pos);
                     emit(*n.requires_clause_expression);
-                    printer.print_cpp2(" )\n\n", n.requires_pos);
+                    printer.print_cpp2(" )\n", n.requires_pos);
                 }
 
                 printer.print_cpp2("class ", n.position());
```

But even without the parsing error, the result cpp1 code is not good (check here: https://cpp2.godbolt.org/z/eGMMz8Txj)
```cpp
#include "cpp2util.h"

requires( true )

class t;
requires( true )

class t {
      public: t() = default;
      public: t(t const&) = delete; /* No 'that' constructor, suppress copy */
      public: auto operator=(t const&) -> void = delete;

};

auto main() -> int;

auto main() -> int{}
```
error:
```
build/_cppfront/main.cpp:5:1: error: expected unqualified-id before 'requires'
    5 | requires( true )
      | ^~~~~~~~
build/_cppfront/main.cpp:8:1: error: expected unqualified-id before 'requires'
    8 | requires( true )
      | ^~~~~~~~
make[2]: *** [CMakeFiles/test.dir/build.make:81: CMakeFiles/test.dir/_cppfront/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/test.dir/all] Error 2
```

--- a/regression-tests/test-results/pure2-requires-clauses.cpp
+++ b/regression-tests/test-results/pure2-requires-clauses.cpp
@@ -11,8 +11,7 @@
 template<typename T, typename U> 
 
     requires( std::is_same_v<T,int> 
-              && std::is_same_v<U,int> )
-class X;
+              && std::is_same_v<U,int> )class X;
 
 //=== Cpp2 type definitions and function declarations ===========================
 
@@ -21,8 +20,8 @@ class X;
 template<typename T, typename U> 
 
     requires( std::is_same_v<T,int> 
-              && std::is_same_v<U,int> )
-class X {
+              && std::is_same_v<U,int> )class X
+ {
     public: explicit X();
 
     public: X(X const&) = delete; /* No 'that' constructor, suppress copy */

--- a/source/cppfront.cpp
+++ b/source/cppfront.cpp
@@ -5094,7 +5094,7 @@ public:
                 if (n.requires_clause_expression) {
                     printer.print_cpp2("requires( ", n.requires_pos);
                     emit(*n.requires_clause_expression);
-                    printer.print_cpp2(" )\n\n", n.requires_pos);
+                    printer.print_cpp2(" )\n", n.requires_pos);
                 }
 
                 printer.print_cpp2("class ", n.position());

--- a/source/parse.h
+++ b/source/parse.h
@@ -6466,14 +6466,20 @@ private:
         }
 
         //  Next is optionally a requires clause
-        if (curr() == "requires") {
-            if (!n->template_parameters) {
-                error("'requires' only valid for a template");
+        if (curr() == "requires")
+        {
+            if (
+                n->is_type()
+                && !n->template_parameters
+                )
+            {
+                error("'requires' is only valid for a type with a template parameter list");
                 return {};
             }
+
             n->requires_pos = curr().position();
             next();
-            auto e = expression();
+            auto e = logical_or_expression();
             if (!e) {
                 error("'requires' must be followed by an expression");
                 return {};

--- a/source/parse.h
+++ b/source/parse.h
@@ -6467,6 +6467,10 @@ private:
 
         //  Next is optionally a requires clause
         if (curr() == "requires") {
+            if (!n->template_parameters) {
+                error("'requires' only valid for a template");
+                return {};
+            }
             n->requires_pos = curr().position();
             next();
             auto e = expression();

